system-parachains/asset-hubs/asset-hub-kusama/src/staking/mod.rs

```diff
@@ -88,35 +97,63 @@ parameter_types! {
 	pub MaxBackersPerWinnerFinal: u32 = MaxElectingVoters::get();
 
 	/// Size of the exposures. This should be small enough to make the reward payouts feasible.
-	pub MaxExposurePageSize: u32 = 512;
+	///
+	/// Safety note: during AHM, this value must be the same as it was on the RC
+	pub const MaxExposurePageSize: u32 = 512;
 }
 
+// Solution type.
+//
+// * Voter index: u32 will scale to a near-infinite number of validators/nominators as npos-voters.
+// While u16 is also enough, it might very well lead to issues if we wish to increase
+// `MaxElectingVoters`. Numbers show that the byte-size of the solution is far from being a
+// bottleneck, ergo using u32.
+// * Target index: 65k is well enough for a network with 1000 validators
+// max.
+// * 24: Note that kusama allows for 24 nominations per nominator.
 frame_election_provider_support::generate_solution_type!(
 	#[compact]
 	pub struct NposCompactSolution24::<
 		VoterIndex = u32,
 		TargetIndex = u16,
 		Accuracy = sp_runtime::PerU16,
-		MaxVoters = MaxElectingVoters,
+		MaxVoters = VoterSnapshotPerBlock,
 	>(24)
 );
 
 parameter_types! {
+	/// AHM Audit: should be the same as RC.
 	pub const BagThresholds: &'static [u64] = &bags_thresholds::THRESHOLDS;
 }
 
+/// We don't want to do any auto-rebags in pallet-bags while the migration is not started or
+/// ongoing.
+pub struct RebagIffMigrationDone;
+impl sp_runtime::traits::Get<u32> for RebagIffMigrationDone {
+	fn get() -> u32 {
+		if cfg!(feature = "runtime-benchmarks") {
+			5
+		} else {
+			if matches!(
+				pallet_ah_migrator::AhMigrationStage::<Runtime>::get(),
```

Comment:
```suggestion
				pallet_ah_migrator::AhMigrationStage::<Runtime>::get().is_done()
```
There should be a function like this, but it would do the same.